### DISCUSSION TITLE
Raspberry pi 4 root privileges problem

### DISCUSSION THREAD

**Discussion by rajivkaushik11** [community/community#122706] [Programming Help]
### Body

I have problems with the root privileges of Raspberry pi 4. I execute my python script using sudo. It executes the script and reaches a function where it requires root privileges. 
The python script needs to be automated and entering password each time is not possible.
I have used the following commands in the sudoers(config file of raspi4) so that it does not ask password:

**username ALL=(ALL) NOPASSWD: ALL
username ALL=(ALL) NOPASSWD: /path/to/script** .These commands did not work as well.

I would like to have some suggestions on this.

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by kazmatw** [comment]
Hi @rajivkaushik11 ✋
To resolve the issue of needing root privileges for running a Python script on a Raspberry Pi without having to enter a password each time, you should modify the sudoers file correctly. It looks like there was a problem with how the sudoers file was edited. Here are step-by-step instructions to correctly configure sudo permissions:

## Step 1: Open the sudoers file safely
The safest way to edit the sudoers file is by using the visudo command, which checks for syntax errors before saving any changes. This prevents saving a corrupted sudoers file that could lock you out from using sudo entirely.

Open a terminal on your Raspberry Pi.
Type sudo visudo and press Enter.
## Step 2: Modify the sudoers file
In the visudo editor, you'll need to add a line that specifies that your user can execute specific commands or all commands without a password. Ensure you're editing under the # User privilege specification section or at the end of the file:

```bash
username ALL=(ALL) NOPASSWD: ALL
```
This line allows the user to run all commands via sudo without entering a password.
If you want to restrict passwordless access to only the Python script:

```bash
username ALL=(ALL) NOPASSWD: /path/to/script.py
```
## Step 3: Save and exit
After adding the necessary line(s), save and exit the editor. If you are using visudo, it typically uses the nano editor by default:
Press Ctrl+X to exit, Y to confirm the changes, and Enter to save.
## Step 4: Test your configuration
Try running your Python script with sudo to see if it asks for a password:
```bash
sudo python3 /path/to/script.py
```
If the script runs without prompting for a password, your sudoers configuration is correct.

Good luck!
